homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,50 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(_: HomeAssistant) -> bool:
+    """Return true as this integration doesn't support any real devices."""
+    return True
+
+
+class DsmrReaderFlowHandler(DiscoveryFlowHandler[Awaitable[bool]], domain=DOMAIN):
+    """Handle DSMR Reader config flow. The MQTT step is inherited from the parent class."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up the config flow."""
+        super().__init__(DOMAIN, "DSMR Reader", _async_has_devices)
+
+    async def async_step_import(self, _: dict[str, Any] | None) -> FlowResult:
+        """Import from configuration.yaml and create config entry."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if not self.hass.services.has_service(domain="mqtt", service="publish"):
+            _LOGGER.warning("DSMR Reader configuration import failed. MQTT is missing")
+            return self.async_abort(reason="mqtt_missing")
+        # There's no configuration supported for this integration, so data can be a fixed object
+        return self.async_create_entry(title="DSMR Reader", data={})
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle user step."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if not self.hass.services.has_service(domain="mqtt", service="publish"):
+            return self.async_abort(reason="mqtt_missing")
+
+        return await self.async_step_confirm()
```

Comment:
Can remove because MQTT check not necessary.

Reply:
```suggestion
```

Reply:
When relying on the implementation in DiscoveryFlowHandler without the MQTT check, users can add the component manually while there's no MQTT server installed. I'd prefer to guide users towards configuring the MQTT server first while installing the component, instead of only doing so in documentation. Because almost all logic in this method is already present in the parent class though, I'm thinking of rewriting this method so the MQTT check is the only functional step done within the method (I'd do almost the same thing in the import method):
```
    async def async_step_user(
        self, user_input: dict[str, Any] | None = None
    ) -> FlowResult:
        """Handle user step."""
        if not self.hass.services.has_service(domain="mqtt", service="publish"):
            return self.async_abort(reason="mqtt_missing")

        return await super().async_step_user(user_input)
```

Reply:
I've removed the mqtt test, as requested. I created issue #75740 to make sure it is documented that an mqtt dependency currently does not trigger an installation when needed.